test/test_intensify/test_intensify.py

```diff
@@ -481,28 +502,42 @@ def target(x):
             deterministic=True, always_race_against=None, run_limit=1)
 
         self.assertEqual(intensifier.n_iters, 0)
+        self.assertEqual(intensifier.stage, IntensifierStage.RUN_FIRST_CONFIG)
 
         # intensification iteration #1
-        # run incumbent first if it was not run before
-        config, _ = intensifier.get_next_challenger(challengers=[self.config2, self.config1],
+        # run first config as incumbent if incumbent is None
+        config, _ = intensifier.get_next_challenger(challengers=[self.config3, self.config2],
                                                     chooser=None)
-        self.assertEqual(config, self.config2)
-        self.assertEqual(intensifier.stage, IntensifierStage.RUN_INCUMBENT)
+        self.assertEqual(config, self.config3)
+        self.assertEqual(intensifier.stage, IntensifierStage.RUN_FIRST_CONFIG)
         # eval config 2 (=first run)
         inc, _ = intensifier.eval_challenger(challenger=config, incumbent=None, run_history=self.rh, )
-        self.assertEqual(inc, self.config2)
-        self.assertEqual(intensifier.stage, IntensifierStage.RUN_CHALLENGER)
+        self.assertEqual(inc, self.config3)
+        self.assertEqual(intensifier.stage, IntensifierStage.RUN_INCUMBENT)
         self.assertEqual(self.stats.inc_changed, 1)
 
+        # regular intensification begins - run incumbent
+        config, _ = intensifier.get_next_challenger(challengers=None,  # since incumbent is run, no configs required
+                                                    chooser=None)
+        self.assertEqual(intensifier.stage, IntensifierStage.RUN_INCUMBENT)
```

Comment:
Isn't this one run too much?

Reply:
Do you mean that this incumbent configuration is evaluated again? It won't be run because it doesn't have any runs left (1 instance & deterministic). So the TA runs remain the same.

Reply:
Okay, and these are tested below. So I guess it's alright then.